sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/init.sql

```diff
@@ -189,13 +190,57 @@ metrics_ping AS (
     )[SAFE_OFFSET(0)] AS adjust_creative,
     ARRAY_AGG(metrics.string.metrics_install_source IGNORE NULLS ORDER BY submission_timestamp ASC)[
       SAFE_OFFSET(0)
-    ] AS install_source
+    ] AS install_source,
+    ARRAY_AGG(
+      metrics.string.metrics_adjust_ad_group IGNORE NULLS
+      ORDER BY
+        submission_timestamp DESC
+    )[SAFE_OFFSET(0)] AS latest_adjust_ad_group,
+    ARRAY_AGG(
+      metrics.string.metrics_adjust_creative IGNORE NULLS
+      ORDER BY
+        submission_timestamp DESC
+    )[SAFE_OFFSET(0)] AS latest_adjust_creative,
+    ARRAY_AGG(
+      metrics.string.metrics_adjust_network IGNORE NULLS
+      ORDER BY
+        submission_timestamp DESC
+    )[SAFE_OFFSET(0)] AS latest_adjust_network,
+    ARRAY_AGG(
+      metrics.string.metrics_adjust_campaign IGNORE NULLS
+      ORDER BY
+        submission_timestamp DESC
+    )[SAFE_OFFSET(0)] AS latest_adjust_campaign,
   FROM
     org_mozilla_firefox.metrics AS org_mozilla_firefox_metrics
   WHERE
     DATE(submission_timestamp) >= '2019-01-01'
   GROUP BY
     client_id
+),
+-- Find most recent client details from the baseline ping.
+baseline_ping AS (
+  SELECT
+    client_info.client_id,
+    DATE(MAX(submission_timestamp)) AS last_seen_date,
+    ARRAY_AGG(normalized_country_code IGNORE NULLS ORDER BY submission_timestamp DESC)[
+      SAFE_OFFSET(0)
+    ] AS last_seen_country,
+    ARRAY_AGG(client_info.device_model IGNORE NULLS ORDER BY submission_timestamp DESC)[
+      SAFE_OFFSET(0)
+    ] AS last_seen_device_model,
+    ARRAY_AGG(client_info.device_manufacturer IGNORE NULLS ORDER BY submission_timestamp DESC)[
+      SAFE_OFFSET(0)
+    ] AS last_seen_device_manufacturer,
+    ARRAY_AGG(client_info.locale IGNORE NULLS ORDER BY submission_timestamp DESC)[
+      SAFE_OFFSET(0)
+    ] AS last_seen_locale,
+  FROM
+    org_mozilla_firefox.baseline AS org_mozilla_firefox_baseline
```

Comment:
Probably? We'd have to add a few `IF ... is_new_profile` expressions to make sure it's equivalent - [query](https://github.com/mozilla/bigquery-etl/blob/08c9352b8048afdb0414919c4284551f5bbab438/sql/moz-fx-data-shared-prod/org_mozilla_firefox/baseline_clients_first_seen/view.sql)?

Reply:
Agree. For the `first_seen` subquery, it requires removing the channel condition and adding a condition for is_new_profile. The benefit is to avoid querying all the clients twice (in two sub-queries) and also avoid scanning all channels to find the release clients.

Reply:
Agreed, I'll do this in a follow-up PR since it won't require schema changes.